### PR TITLE
[Keyboard, iOS] Use special key mapping for logical keys

### PR DESCRIPTION
**Pull request by dkwingsmt**
This PR fixes the logic to derive logical keys by checking a "special key mapping" first, fixing https://github.com/flutter/flutter/issues/106407.

This logic has long been applied to `RawKeyboardIos` ([here](https://github.com/flutter/flutter/blob/47be307f51f01bbf5af82ac9d8303e71710f4085/packages/flutter/lib/src/services/raw_keyboard_ios.dart#L110)) but was somehow missing in the embedding implementation. The iOS key event uses some literal strings such as `"UIKeyInputEscape"` as `characters` for certain special keys.

## Pre-launch Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I signed the [CLA].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: shell/platform/darwin/ios/framework/Source/FlutterEmbedderKeyResponder.mm [hunk 1]
```diff
@@ -150,6 +150,12 @@ static uint64_t toLower(uint64_t n) {
  */
 static uint64_t GetLogicalKeyForEvent(FlutterUIPressProxy* press, uint64_t physicalKey)
     API_AVAILABLE(ios(13.4)) {
+  // Some unprintable keys on iOS has literal names on their key label, such as
```
Comment (gspencergoog) [inline shell/platform/darwin/ios/framework/Source/FlutterEmbedderKeyResponder.mm:153]:
```suggestion
  // Some unprintable keys on iOS have literal names on their key label, such as
```

**Review by gspencergoog — APPROVED** [review]
32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d

**Comment by dupuchba** [comment]
Minor precision on the label, it's not only the `text-input` that is affected but the whole application. If you press escape (cmd+.) it will crash all apps even without any text-field

**Comment by dkwingsmt** [comment]
> Minor precision on the label, it's not only the `text-input` that is affected but the whole application. If you press escape (cmd+.) it will crash all apps even without any text-field

Don't worry, I'm just marking everything related to keyboard with `text-input` for me to easily group.

**Pull request merged**